src/main/java/com/peerrequest/app/api/DirectRequestsController.java

```diff
@@ -0,0 +1,264 @@
+        if (this.entryService.get(entryId).isEmpty()) {
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND, "entry does not exist");
+        }
+
+        if (dto.openSlots().isEmpty()) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "open slots is required");
+        }
+
+        String researcherId = this.entryService.get(entryId).get().getResearcherId();
+
+        if (!user.getAttribute("sub").toString().equals(researcherId)) {
+            throw new ResponseStatusException(HttpStatus.FORBIDDEN,
+                    "Only the user that created the entry may create a request process");
+        }
+        // TODO: check if there is an existent Bidding process with the entry
+        var directRequestProcessDto = DirectRequestProcess.fromDto(dto, entryId);
+        return this.directRequestProcessService.create(directRequestProcessDto.toDto()).toDto();
+    }
+
+    /**
+     * Patches a directRequestProcess.
+     *
+     * @param entryId entry of the directRequestProcess
+     * @param dto updater containing the amount of openSlots (int)
+     *
+     * @return updated directRequestProcess
+     */
+    @PutMapping(value = "/categories/{category_id}/entries/{entry_id}/process", produces = "application/json")
+    public DirectRequestProcess.Dto patchDirectRequestProcess(@PathVariable("entry_id") final Long entryId,
+                                                          @RequestBody final DirectRequestProcess.Dto dto,
+                                                          @AuthenticationPrincipal OAuth2User user) {
+        if (this.entryService.get(entryId).isEmpty()) {
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND, "entry does not exist");
+        }
+
+        String researcherId = this.entryService.get(entryId).get().getResearcherId();
+        if (!user.getAttribute("sub").toString().equals(researcherId)) {
+            throw new ResponseStatusException(HttpStatus.FORBIDDEN,
+                    "Only the user that created the entry may create a request process");
+        }
+
+        if (dto.openSlots().isEmpty()) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "open slots must be given");
+        }
+
+        if (dto.openSlots().get() < 0) {
+            throw new ResponseStatusException(HttpStatus.I_AM_A_TEAPOT, "open slots value must be greater than 0");
+        }
+
+        var option = this.directRequestProcessService.update(entryId, dto);
+        if (option.isEmpty()) {
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND, "direct request process does not exist");
+        }
+
+        return option.get().toDto();
+    }
+
+    /**
+     * Gets a directRequests.
+     *
+     * @param requestId request to get
+     *
+     * @return directRequest with requestId
+     */
+    @GetMapping(value = "/categories/{category_id}/entries/{entry_id}/process/requests/{request_id}",
```

Comment:
remove:  produce = "aplication/json"
to match the other controllers

Reply:
Done.